tensorflow/core/kernels/scan_ops.cc

```diff
@@ -104,12 +104,15 @@ namespace functor {
   DECLARE(Eigen::internal::ProdReducer<T>, T);
 
 TF_CALL_GPU_NUMBER_TYPES(DECLARE_FOR_ALL_REDUCERS);
+TF_CALL_bfloat16(DECLARE_FOR_ALL_REDUCERS);
 DECLARE_FOR_ALL_REDUCERS(int32);
 DECLARE_FOR_ALL_REDUCERS(int64_t);
 #undef DECLARE_FOR_ALL_REDUCERS
 
 #define DECLARE_FOR_LOGSUMEXP_REDUCER(T) DECLARE(LogSumExpReducer<T>, T);
-TF_CALL_GPU_NUMBER_TYPES(DECLARE_FOR_LOGSUMEXP_REDUCER)
+TF_CALL_half(DECLARE_FOR_LOGSUMEXP_REDUCER);
```

Comment:
Thanks @cantonios for the review!

I will be adding `bfloat16` to `TF_CALL_GPU_NUMBER_TYPES` in a future PR, so I didn't want that change to add bf16 to `CumulativeLogsumexp` since it wasn't allowed by the op definition.

However, if I add bf16 to `CumulativeLogsumexp` as per your previous comment, I can change these back.

Reply:
Yes, can you please do this?

Reply:
@cantonios I've added BF16 support for `CumulativeLogsumexp`.